src/cljs/lomake_editori/editor/component.cljs

```diff
@@ -0,0 +1,112 @@
+           #_[language lang]])))))
+
+(defn link-info [{:keys [params] :as content} path]
+  (let [languages (subscribe [:editor/languages])
+        value     (subscribe [:editor/get-component-value path])]
+    (fn [{:keys [params] :as content} path]
+      (into
+        [:div.link-info
+         [:p "Linkki"]]
+        (for [lang @languages]
+          [:div
+           [:p "Osoite"]
+           [:input {:value       (get-in @value [:params :href lang])
+                    :type        "url"
+                    :on-change   #(dispatch [:editor/set-component-value (-> % .-target .-value) path :params :href lang])
+                    :placeholder "http://"}]
+           [language lang]
+           [:input {:on-change   #(dispatch [:editor/set-component-value (-> % .-target .-value) path :text lang])
+                    :value       (get-in @value [:text lang])
+                    :placeholder "Otsikko"}]
+           [language lang]])))))
+
+(defn info [{:keys [params] :as content} path]
+  (let [languages (subscribe [:editor/languages])
+        value     (subscribe [:editor/get-component-value path :text])]
+    (fn [{:keys [params] :as content} path]
+      (into
+        [:div.info
+         [:p "Ohjeteksti"]]
+        (for [lang @languages]
+          [:div
+           [:input
+            {:value       (get @value lang)
+             :on-change   #(dispatch [:editor/set-component-value (-> % .-target .-value) path :text lang])
+             :placeholder "Ohjetekstin sisältö"}]
+           [language lang]
+           ])))))
+
+(defn ^:private delayed-trigger [timeout-ms on-trigger]
+  (let [latch (atom nil)]
+    [(fn [] (reset! latch nil))
+     (fn [event]
+       (let [local-latch (atom nil)
+             handle      (js/setTimeout (fn []
+                                          (when (= @latch @local-latch)
+                                            (on-trigger event)))
+                           timeout-ms)]
+         (do
+           (reset! local-latch handle)
+           (reset! latch handle))))]))
+
+(def ^:private toolbar-elements
+  (let [dummy [:div "ei vielä toteutettu.."]]
+    {"Lomakeosio"                component/form-section
+     "Tekstikenttä"              component/text-field}))
+;"Tekstialue"                dummy
+;"Lista, monta valittavissa" dummy
+;"Lista, yksi valittavissa"  dummy
+;"Pudotusvalikko"            dummy
+;"Vierekkäiset kentät"       dummy
+;"Liitetiedosto"             dummy
+;"Ohjeteksti"                info
+;"Linkki"                    link-info
+;"Väliotsikko"               dummy
+
+(defn ^:private component-toolbar [path]
+  (fn [path]
+    (into [:ul]
+      (for [[component-name generate-fn] toolbar-elements]
+        [:li {:on-click #(dispatch [:generate-component generate-fn path])}
+         component-name]))))
+
+(defn add-component [path]
+  (let [show-bar? (reaction nil)
```

Comment:
This is a bit confusing. reaction would seem to indicate that we are dealing with re-frame app's state. But we use show-bar? as local state instead. Can we use reagent's atom (reagent.core/atom) here instead? It would make the local scope here more explicit.